sdk/build.gradle.kts

```diff
@@ -73,6 +73,7 @@ dependencies {
   api(project(":plugin-viewport"))
   compileOnly(Dependencies.asyncInflater)
   implementation(Dependencies.kotlin)
+  implementation(Dependencies.coroutines)
```

Comment:
do you have data on how much this increases our sdk size?

Reply:
Good question, I will try to measure it.  But I guess we should not be afraid of using language features.

Reply:
What about method count? I'm also in doubt whether we need to add dependency on coroutines for one simple call

Reply:
btw, it's base module. If user use our sdk only with several module it might increase binary size.

>At least the debug builds are seem to be almost exactly the same size, with or without coroutines
I guess because we already have `coroutines` in some module

Reply:
In order to avoid adding coroutine dependency, we could use something like this instead:

```
ExecutorService executor = Executors.newSingleThreadExecutor();
Handler handler = new Handler(Looper.getMainLooper());

executor.execute(new Runnable() {
    @Override
    public void run() {
        ....
    }
}

Reply:
@yunikkk @alexander-kulikovskii would you be happier with `executor`?

Reply:
or even
```
new Thread(new Runnable() {
    @Override
    public void run() {
        ....
    }
}

Reply:
I don't think `executor` is needed. You can [post](https://developer.android.com/reference/android/os/Handler#post(java.lang.Runnable)) to the `handler` directly.

Reply:
IIRC, your coroutine solution was just enqueuing the _code_ to be run in the main thread later on. No need for a bg thread in the Handler solution either if you don't want to explicitly run in a bg thread.

Reply:
No, we don't want/need it on the main thread, we are just sending an event for turnstile so the user can be billed, nothing to do in the UI.

Reply:
Ok. Then, you could use the kotlin extension function https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.concurrent/thread.html:
```Kotlin
thread {
    mapTelemetry.onAppUserTurnstileEvent()
}
```

Sidenote: is there already any other bg thread we could re-use instead of creating a new one?

Reply:
@paulrenn67 I'm happy with either thread/executor. 
What I would also suggest is to file a ticket in common to dispatch this call to the BG instead of the client (then our other libs using it will also benefit from the speedup)

Reply:
btw, Kotlin coroutines seems to add around 100kB post-shrinking. (Seen in [AndroidX API Guidelines](https://cs.android.com/androidx/platform/frameworks/support/+/androidx-main:docs/api_guidelines/dependencies.md;bpv=0))